processing/src/main/java/io/druid/segment/DoubleDimensionHandler.java

```diff
@@ -19,17 +19,23 @@
 
 package io.druid.segment;
 
-import io.druid.segment.column.Column;
 import io.druid.segment.column.ColumnCapabilities;
-import io.druid.segment.column.GenericColumn;
-import io.druid.segment.data.Indexed;
+import io.druid.segment.selector.settable.SettableColumnValueSelector;
+import io.druid.segment.selector.settable.SettableDoubleColumnValueSelector;
 import io.druid.segment.writeout.SegmentWriteOutMedium;
 
-import java.io.Closeable;
-import java.io.IOException;
+import java.util.Comparator;
 
 public class DoubleDimensionHandler implements DimensionHandler<Double, Double, Double>
 {
+  /**
+   * Don't use {@link Comparator#comparingDouble} because this comparator is used in hot code, so avoiding small extra
+   * indirection.
+   */
+  @SuppressWarnings("ComparatorCombinators")
+  private static Comparator<ColumnValueSelector> DOUBLE_COLUMN_COMPARATOR =
+      (s1, s2) -> Double.compare(s1.getDouble(), s2.getDouble());
```

Comment:
getDouble can throw assertion error for Nulls. need to handle that case here.

Reply:
Thanks for this finding, fixed.